File: agent-dao/src/main/java/com/agent/platform/dao/entity/sys/SysOperLog.java
package com.agent.platform.dao.entity.sys;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 操作日志（由 Controller 切面自动记录关键写操作）
 */
@Data
@TableName("sys_oper_log")
public class SysOperLog {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long tenantId;

    /** 操作人ID */
    private Long userId;

    /** 操作人登录名 */
    private String username;

    /** 所属模块 */
    private String module;

    /** 操作内容 */
    private String operation;

    /** HTTP 方法 */
    private String method;

    /** 请求路径 */
    private String uri;

    /** 来源IP */
    private String ip;

    /** 是否成功：0失败 1成功 */
    private Integer success;

    /** 错误信息 */
    private String errorMsg;

    /** 耗时(毫秒) */
    private Integer costMs;

    private LocalDateTime createTime;
}

File: frontend/src/api/sys-oper-log.ts
import request from './request'
import type { PageResult } from './types'

/** 操作日志 */
export interface OperLog {
  id: number
  tenantId: number
  userId?: number
  username?: string
  module?: string
  operation?: string
  method?: string
  uri?: string
  ip?: string
  /** 0失败 1成功 */
  success: number
  errorMsg?: string
  costMs?: number
  createTime?: string
}

export interface OperLogQuery {
  page?: number
  size?: number
  keyword?: string
  module?: string
  success?: number
  startTime?: string
  endTime?: string
}

/** 操作日志 API（/api/sys/oper-logs） */
export const operLogApi = {
  page(params: OperLogQuery) {
    return request.get<never, PageResult<OperLog>>('/sys/oper-logs', { params })
  }
}
